plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/runtime/CloudSdkRuntimeTest.java

```diff
@@ -0,0 +1,104 @@
+package com.google.cloud.tools.eclipse.appengine.localserver.runtime;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Field;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.wst.server.core.internal.Runtime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CloudSdkRuntimeTest {
+  
+  @Mock
+  private Runtime runtime;
+  @Mock
+  private IPath mockLocation;
+  
+  @Test(expected = NullPointerException.class)
+  public void testValidate_withoutRuntime() {
+    new CloudSdkRuntime().validate();
+  }
+  
+  @Test
+  public void testValidate_withRuntimeWithoutName() throws Exception {
+    CloudSdkRuntime cloudSdkRuntime = new CloudSdkRuntime();
+    setField(cloudSdkRuntime, "runtime", runtime);
+    cloudSdkRuntime.validate();
+  }
+
+  @Test
+  public void testValidate_withRuntimeWithEmptyName() throws Exception {
+    CloudSdkRuntime cloudSdkRuntime = new CloudSdkRuntime();
+    setField(cloudSdkRuntime, "runtime", runtime);
+    when(runtime.getName()).thenReturn("");
+    IStatus status = cloudSdkRuntime.validate();
+    assertThat(status.getSeverity(), is(IStatus.ERROR));
+    assertThat(status.getMessage(), is(org.eclipse.wst.server.core.internal.Messages.errorRuntimeName));
```

Comment:
better to use a literal string if it avoids a dependency, unless we expect this to change